src/software/thunderscope/thunderscope_main.py

```diff
@@ -403,7 +395,24 @@ def __async_sim_ticker(tick_rate_ms: int) -> None:
             args.blue_full_system_runtime_dir,
         ) as blue_logger, ProtoLogger(
             args.yellow_full_system_runtime_dir,
-        ) as yellow_logger, Gamecontroller() as gamecontroller:
```

Comment:
We should pass a custom `time_provider` to `ProtoLogger` when running in CI mode so the replay timing is correct. This would be similar to how the simulated test fixtures are setup:

https://github.com/UBC-Thunderbots/Software/blob/0d8d38136a4452c102bb3de599b1e9bc9c129339/src/software/field_tests/field_test_fixture.py#L421-L433

Currently, watching the AI vs AI replay from CI doesn't run in realtime.

Reply:
ah I noticed this too but didnt realize we had a way of providing the time directly to the proto logger. nice catch.